File: 2.Practice/Arrays/Easy/1260.Shift2DGrid/python/solution2.py
class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        rows = len(grid)
        cols = len(grid[0])
        lst = []
        for row in grid:
            lst.extend(row)
        items = deque(lst)

        # k %= len(items)

        items.rotate(k)
        print(items)
        items = list(items)
        lst = []
        for i in range(0, len(items), cols):
            lst.append(items[i:i+cols])
        return lst

File: 2.Practice/Arrays/Easy/1260.Shift2DGrid/js/solution2.js
/**
 * @param {number[][]} grid
 * @param {number} k
 * @return {number[][]}
 */
var shiftGrid = function(grid, k) {
    let cols = grid[0].length;
    let rows = grid.length;
    let len = cols * rows;
    let items = [];
    let newGrid = [];
    for (let item of grid) {
        items = items.concat(item);
    }

    let step = 0;
    k = k % len;
    while (step < k) {
        let last = items.splice(len - 1, 1)[0];
        items.unshift(last);
        step++;
    }
    for (let i = 0; i < rows; i++) {
        newGrid.push([]);
        for (let j = 0; j < cols; j++) {
            newGrid[i].push(items[i * cols + j]);
        }
    }


    return newGrid;
};